src/MoBi.Presentation/Presenter/EditSumFormulaPresenter.cs

```diff
@@ -38,15 +53,58 @@ public override void Edit(SumFormula sumFormula)
 
       private void refreshView()
       {
-         _sumFormulaDTO = _sumFormulaToDTOSumFormulaMapper.MapFrom(_formula);
+         _sumFormulaDTO = _sumFormulaDTOMapper.MapFrom(_formula);
+         _editFormulaPathListPresenter.Edit(_formula);
          _descriptorConditionListPresenter.Edit(_formula, x => x.Criteria, BuildingBlock);
+         Validate(_formula.FormulaString);
          _view.Show(_sumFormulaDTO);
       }
 
-      public void ChangeVariableName(string newVariableName, string oldVariableName)
+      public void ChangeVariableName(string newVariableName)
+      {
+         AddCommand(_moBiFormulaTask.ChangeVariableName(_formula, newVariableName, BuildingBlock));
+      }
+
+      public void Validate(string formulaString)
+      {
+         var (_, message) = _moBiFormulaTask.Validate(formulaString, _formula, BuildingBlock);
```

Comment:
only validation _message_ importnat here but not the validation _result_?

Reply:
Nope. No error, message empty. Clear the view. I don't know why the API is different and I would do it differently if I had time.